### ISSUE TITLE
Handle Agreement tag

### ISSUE THREAD

**Issue by cassidyjames** [elementary/appcenter#729]
There's now an agreement tag in Appstream. This is distinct from a code license, where it's an agreement a user must agree to before using the app. While I don't anticipate many apps utilizing this in AppCenter, other repo apps may, and it is probably worth implementing for the spec.

My initial idea was a dialog when installing.

See: https://github.com/ximion/appstream/issues/188#issuecomment-385075318

**Comment by cassidyjames** [comment]
This would be inline instead of a link, but here's a quick sketch: ![sketch](https://i.imgur.com/Vgt5b1p.jpg)

**Comment by danrabbit** [comment]
Should this maybe be something like `dialog-question` and/or introducing a color License icon instead of a warning?

[From the spec](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-agreement) it looks like this is meant to show the full text of the agreement, possibly divided into sections, rather than a link.

I think this is also something different from a "proprietary license". This could be something like "You agree not to upload hate speech using this app or we reserve the right to terminate your access" or something right?

We might want something closer to the below where you have a scrollbox and we make the "Accept" button sensitive when you hit the bottom of the box:

![](https://cdn.dribbble.com/users/160094/screenshots/3313644/day089.gif)

**Comment by danrabbit** [comment]
This mockup is interesting to me because it indicates the last date that the ToS was updated in the secondary text:

![](https://cdn.dribbble.com/users/1429493/screenshots/4214168/terms-01.jpg)

**Comment by cassidyjames** [comment]
@danrabbit yeah the scroll box makes sense, and a more generic/less scary license icon would be appropriate. I don't see an updated date in the spec, just a version string. So I'm not sure we can do that, though I like the idea.
